core/wiki/macros/toolbar.tid

```diff
@@ -0,0 +1,73 @@
+title: $:/core/macros/toolbar
+tags: $:/tags/Macro
+
+\define toolbar-filter-display(tag,config)
+<$set name="config" filter="[<__config__>]match[]]" 
+		value="[all[shadows+tiddlers]tag[$tag$]!has[draft.of]]" 
+		emptyValue="[all[shadows+tiddlers]tag[$tag$]!has[draft.of]] :filter[lookup:show[$config$]!match[hide]]" >
+	<<config>>
+</$set>
+\end
+
+\define basic-buttons(tag,config,class)
+\whitespace trim
+<$list filter=<<toolbar-filter-display "$tag$" "$config$">> variable="listItem">
+	<$set name="tv-config-toolbar-class" filter="[<tv-config-toolbar-class>] [<listItem>encodeuricomponent[]addprefix[tc-btn-]] [<__class__>]" >
+		<$transclude tiddler=<<listItem>>/>
+	</$set>
+</$list>
+\end
+
+\define buttons-with-template(tag,config,class,template)
+\whitespace trim
+<$list filter=<<toolbar-filter-display "$tag$" "$config$">> >
+	<$set name="tv-config-toolbar-class" filter="[<tv-config-toolbar-class>] [<listItem>encodeuricomponent[]addprefix[tc-btn-]] [<__class__>]" >
+		<$transclude tiddler="$template$" />
+	</$set>
+</$list>
+\end
+
+\define toolbar(tag,config,class,template)
+\whitespace trim
+<$set name="buttons" 
+		filter="[<__template__>!match[]]" 
+		value="""<$macrocall $name="buttons-with-template" tag=<<__tag__>> config= <<__config__>> class=<<__class__>> template=<<__template__>> />""" 
+		emptyValue="""<$macrocall $name="basic-buttons" tag=<<__tag__>> config= <<__config__>> class=<<__class__>> />""" >
+	<<buttons>>
+</$set>
+\end
+
+
+\define toolbar-filter-drop-down(tag,config)
+[all[shadows+tiddlers]tag[$tag$]!has[draft.of]] :filter[lookup:show[$config$]match[hide]]
+\end
+
+\define toolbar-basic-drop-down-list(tag,config,class)
```

Comment:
whitespace trim is missing

Reply:
It is necessary? If this macro (and the following macro) is always used inside other macro with this pragma. I don't think that someone uses the macro `toolbar-basic-drop-down-list` directly.

I think I forgot to remove it on the lines 13 and 22.